indico/modules/events/editing/controllers/backend/timeline.py

```diff
@@ -379,8 +379,13 @@ def _process(self):
                 filename = secure_filename(file.filename, f'file-{file.id}')
                 file_type = revision_file.file_type
                 folder_name = secure_filename(file_type.name, f'file-type-{file_type.id}')
+                first_revision = (EditingRevisionFile.query
+                                  .filter_by(file=file)
+                                  .join(EditingRevision)
+                                  .order_by(EditingRevision.created_dt)
+                                  .first().revision)
```

Comment:
```suggestion
                first_revision = revision_file.revision.editable.revisions[0]
```

However, in my test I somehow got slightly different results:

```
In [23]: rf = EditingRevisionFile.query[-1]

In [24]: rf
Out[24]: <EditingRevisionFile(revision_id=760, file_id=6579)>

In [25]: rf.revision.editable.revisions[0]
Out[25]: <EditingRevision(714, editable_id=310, ready_for_review)>

In [26]: (EditingRevisionFile.query
    ...:                     .filter_by(file=rf.file)
    ...:                     .join(EditingRevision)
    ...:                     .order_by(EditingRevision.created_dt)
    ...:                     .first().revision)
Out[26]: <EditingRevision(715, editable_id=310, ready_for_review)>

In [27]: rf.revision.editable.revisions
Out[27]:
[<EditingRevision(714, editable_id=310, ready_for_review)>,
 <EditingRevision(715, editable_id=310, ready_for_review)>,
 <EditingRevision(716, editable_id=310, ready_for_review)>,
 <EditingRevision(717, editable_id=310, ready_for_review)>,
 <EditingRevision(974, editable_id=310, needs_submitter_changes)>,
 <EditingRevision(718, editable_id=310, ready_for_review)>,
 <EditingRevision(719, editable_id=310, ready_for_review)>,
 <EditingRevision(720, editable_id=310, ready_for_review)>,
 <EditingRevision(760, editable_id=310, needs_submitter_confirmation)>,
 <EditingRevision(1549, editable_id=310, rejection)>]
```

Most likely that's because the first revision does not contain this particular file.

So the main question here is what we want: The date of the first revision (my suggestion) or the date of the first revision that added a file (your code).

In the latter case you could also just use this instead of a custom query:

```py
first_revision = min(file.editing_revision_files, key=attrgetter('revision.created_dt')).revision
```

Reply:
I don't think it makes any sense to use the first revision of the editable. We just want to figure out the dt of the revision that added the file (the dt of the File itself isn't ideal because the time of the upload differs a bit from the submission time, and can be misleading)

your later suggestion looks good tho, I'll give it a try